main.js

```diff
@@ -0,0 +1,162 @@
+var _startX = 0;            // mouse starting positions
+var _startY = 0;
+var _offsetX = 0;           // current element offset
+var _offsetY = 0;
+var _dragElement;           // needs to be passed from OnMouseDown to OnMouseMove
+var _oldZIndex = 0;         // we temporarily increase the z-index during drag
+var iterator = 10;
+
+var premadePhrases = {
+    sample:"the,quick,brown,fox,jumps,over,the,lazy,dog",
+    lolcat:"lol,cat,lolcat,iz,r,ur,good,?,i,want,cheezburger,nao",
+    geek:"battle,awesome,are,an,action,could,complete,cheat,bit,force,information,is",
+    bacon:"the,day,begin,s,with,this,morning,marvel,bring,ing,the,aroma,of,satisfaction,bacon",
+    beard:"my,beard,is,a,lifestyle,facial,hair,touch,it,thanks",
+    explicit:"fuck,fuck,fuck"
+};
+
+
+function initDragDrop() {
+    document.addEventListener('mousedown', onMouseDown);
+}
+
+initDragDrop();
+
+function onMouseDown(e) {
+    // IE doesn't pass the event object
+    if (e == null) 
+        e = window.event; 
+    
+    // IE uses srcElement, others use target
+    var target = e.target != null ? e.target : e.srcElement;
+    
+    // for IE, left click == 1
+    // for Firefox, left click == 0
+    if ((e.button == 1 && window.event != null || e.button == 0) && target.className.indexOf('drag') > -1) {
+        // grab the mouse position
+        _startX = e.clientX;
+        _startY = e.clientY;
+        
+        // grab the clicked element's position
+        _offsetX = extractNumber(target.style.left);
+        _offsetY = extractNumber(target.style.top);
+        
+        // bring the clicked element to the front while it is being dragged
+        iterator++;
+        target.style.zIndex = iterator;
```

Comment:
I know I helped you with this z-index solution but now I don't like it. If you tried to do something like make the door handle always on top then this makes it difficult. I think a good solution would be to remove all z-index's on the magnets and on mousedown, remove the magnet from the dom then append it back so the natural flow of the dom will make it appear on top of them all.